### ISSUE TITLE
Meta: `onyo_get` support for directories and templates

### ISSUE THREAD

**Issue by aqw** [psyinfra/onyo#673]
This is a tracking, meta issue to allow `onyo get` to select the type of data source to query (other than just assets).

The objective is to add support for directories and templates. This result would obviate the need for an `onyo list` (#53), `onyo template` (#552), and address #672.

- [ ] #676 (default to `asset`; the present behavior)
- [ ] #674
- [ ] #675

---

Open Questions:
- what are the implications for `onyo set` and `onyo unset`
  - they both exclusively operate on assets (which is very, very sane)
  - but if one can query `template` keys, one naturally expects that one can `set` them...

**Comment by bpoldrack** [comment]
Just adding another open question:

Directories could be "matchable" in various ways. They do have some metadata themselves (empty, not empty, glob, creation/modification time, etc.)
So, how is `--match` interpreted, when `--type=directory`?

- not allow it at all?
- have special expressions that `--match` would accept in that mode: some notion of  `contains`  (including `no asset`)?
- match assets within via `--match` and have a separate option to be used for directory matching

**Comment by bpoldrack** [comment]
> but if one can query template keys, one naturally expects that one can set them...

Seems valid to me. I think this would suggest to add a `--type=template|asset` to `set`, `unset` and potentially even `new`.
I'd give that low priority ATM, though.

**Comment by nhjjr** [comment]
I'm in favor of this; I missed this recently when adding devices but wasn't sure what the templates were called. It's of course solved with `ls .onyo/templates` but that's only because I already know where the folder is located.

**Comment by aqw** [comment]
Yeah, tab completion is meant to alleviate this somewhat when creating assets with `onyo new`. But I agree that it's an incomplete solution.
